Allow the default log location to be configurable (e.g. /var/log),
and use separate logs for each display instance (e.g. XWin.0.log).

---
 configure.ac             |    1 +
 hw/xwin/InitOutput.c     |   23 ++++++++++++++---------
 hw/xwin/winglobals.c     |    4 ++--
 hw/xwin/winprocarg.c     |    4 ----
 include/xwin-config.h.in |    1 -
 5 files changed, 17 insertions(+), 16 deletions(-)

--- origsrc/xorg-server-1.5.3/configure.ac	2008-11-12 21:54:18.257937400 -0600
+++ src/xorg-server-1.5.3/configure.ac	2008-11-12 22:13:01.507937400 -0600
@@ -1659,6 +1659,7 @@
 AC_MSG_RESULT([$XWIN])
 
 if test "x$XWIN" = xyes; then
+	AC_DEFINE_DIR(DEFAULT_LOGDIR, logdir, [Default log location])
 	AC_DEFINE_UNQUOTED(XORG_VERSION_CURRENT, [$VENDOR_RELEASE], [Current Xorg version])
 	AC_DEFINE_UNQUOTED(__VENDORDWEBSUPPORT__, ["$VENDOR_WEB"], [Vendor web address for support])
 	AC_CHECK_TOOL(WINDRES, windres)
--- origsrc/xorg-server-1.5.3/hw/xwin/InitOutput.c	2008-11-12 21:54:13.945437400 -0600
+++ src/xorg-server-1.5.3/hw/xwin/InitOutput.c	2008-11-12 22:31:23.097847400 -0600
@@ -657,19 +657,25 @@
         if (shfolder != NULL)
             FreeLibrary(shfolder);
     }
+#endif /* RELOCATE_PROJECTROOT */
     if (!g_fLogFileChanged) {
         static char buffer[MAX_PATH];
-        DWORD size = GetTempPath(sizeof(buffer), buffer);
+        DWORD size;
+#ifdef RELOCATE_PROJECTROOT
+        size = GetTempPath(sizeof(buffer), buffer);
+#else
+        size = snprintf(buffer, sizeof(buffer), DEFAULT_LOGDIR"/");
+#endif
         if (size && size < sizeof(buffer))
         {
             snprintf(buffer + size, sizeof(buffer) - size, 
-                    "XWin.%s.log", display); 
+                    "XWin.%s.log", display);
             buffer[sizeof(buffer)-1] = 0;
             g_pszLogFile = buffer;
             winMsg (X_DEFAULT, "Logfile set to \"%s\"\n", g_pszLogFile);
         }
     }
-#ifdef XKB
+#if defined(RELOCATE_PROJECTROOT) && defined(XKB)
     {
         static char xkbbasedir[MAX_PATH];
 
@@ -679,8 +685,7 @@
         XkbBaseDirectory = xkbbasedir;
 	XkbBinDirectory = basedir;
     }
-#endif /* XKB */
-#endif /* RELOCATE_PROJECTROOT */
+#endif
 }
 
 void
@@ -888,7 +888,7 @@
 #endif
 
   ErrorF ("-logfile filename\n"
-	  "\tWrite logmessages to <filename> instead of /tmp/Xwin.log.\n");
+	  "\tWrite logmessages to <filename>.\n");
 
   ErrorF ("-logverbose verbosity\n"
 	  "\tSet the verbosity of logmessages. [NOTE: Only a few messages\n"
@@ -926,9 +926,9 @@
   /* Notify user where UseMsg text can be found.*/
   if (!g_fNoHelpMessageBox)
     winMessageBoxF ("The " PROJECT_NAME " help text has been printed to "
-		  "/tmp/XWin.log.\n"
-		  "Please open /tmp/XWin.log to read the help text.\n",
-		  MB_ICONINFORMATION);
+		  "%s\n"
+		  "Please open %s to read the help text.\n",
+		  MB_ICONINFORMATION, g_pszLogFile, g_pszLogFile);
 }
 
 /* ddxInitGlobals - called by |InitGlobals| from os/util.c */
--- origsrc/xorg-server-1.5.3/hw/xwin/winglobals.c	2008-11-18 00:47:16.495683100 -0600
+++ src/xorg-server-1.5.3/hw/xwin/winglobals.c	2008-11-18 00:47:24.730058100 -0600
@@ -64,11 +64,11 @@
 HICON		g_hIconX = NULL;
 HICON		g_hSmallIconX = NULL;
 #ifndef RELOCATE_PROJECTROOT
-char *		g_pszLogFile = "/tmp/XWin.log";
+char *		g_pszLogFile = DEFAULT_LOGDIR "/XWin.log";
 #else
 char *		g_pszLogFile = "XWin.log";
-Bool		g_fLogFileChanged = FALSE;
 #endif
+Bool		g_fLogFileChanged = FALSE;
 int		g_iLogVerbose = 2;
 Bool		g_fLogInited = FALSE;
 char *		g_pszCommandLine = NULL;
--- origsrc/xorg-server-1.5.3/hw/xwin/winprocarg.c	2008-11-18 00:15:43.591088100 -0600
+++ src/xorg-server-1.5.3/hw/xwin/winprocarg.c	2008-11-18 00:45:13.886308100 -0600
@@ -54,9 +54,7 @@
 #endif
 extern int			g_iLogVerbose;
 extern char *			g_pszLogFile;
-#ifdef RELOCATE_PROJECTROOT
 extern Bool			g_fLogFileChanged;
-#endif
 extern Bool			g_fXdmcpEnabled;
 extern Bool			g_fAuthEnabled;
 extern char *			g_pszCommandLine;
@@ -1366,9 +1364,7 @@
     {
       CHECK_ARGS (1);
       g_pszLogFile = argv[++i];
-#ifdef RELOCATE_PROJECTROOT
       g_fLogFileChanged = TRUE;
-#endif
       return 2;
     }
 
--- origsrc/xorg-server-1.5.3/include/xwin-config.h.in	2008-11-12 21:54:18.273562400 -0600
+++ src/xorg-server-1.5.3/include/xwin-config.h.in	2008-11-12 22:15:07.195437400 -0600
@@ -24,4 +24,7 @@
 #undef NEED_SNPRINTF
 
 /* Vendor web address for support */
-#undef __VENDORDWEBSUPPORT__
\ No newline at end of file
+#undef __VENDORDWEBSUPPORT__
+
+/* Default log location */
+#undef DEFAULT_LOGDIR
